pagerduty/user.go

```diff
@@ -200,7 +204,22 @@ func (s *UserService) Create(user *User) (*User, *Response, error) {
 
 	resp, err := s.client.newRequestDo("POST", u, nil, &UserPayload{User: user}, &v)
 	if err != nil {
-		return nil, nil, err
+		if e, ok := err.(*Error); ok && strings.Compare(fmt.Sprintf("%v", e.Errors), "[Email has already been taken]") == 0 {
+			resp, _, lErr := s.List(&ListUsersOptions{Query: user.Email})
+			if lErr != nil {
+				return nil, nil, fmt.Errorf("user email address is not unique and failed to fetch existing users: %w", lErr)
+			}
+
+			for _, u := range resp.Users {
+				if isSameUser(u, user) {
+					return s.Get(u.ID, &GetUserOptions{})
```

Comment:
Maybe you do not want to return here yet. Note there is a caching added below. I think just a beak of the loop should be cleaner.